images/webhooks/src/validators/mcValidator.go

```diff
@@ -0,0 +1,141 @@
+package validators
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/go-logr/logr"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"webhooks/funcs"
+	"webhooks/linstor"
+)
+
+type mc struct {
+	Spec     mcSpec   `json:"spec"`
+	Metadata Metadata `json:"metadata"`
+}
+
+type mcSpec struct {
+	Settings mcSettings `json:"settings"`
+}
+
+type mcSettings struct {
+	DrbdPortRange drbdPortRange `json:"drbdPortRange"`
+}
+
+type drbdPortRange struct {
+	MinPort *int `json:"minPort" `
+	MaxPort *int `json:"maxPort"`
+}
+
+const (
+	propContainerName = "d2ef39f4afb6fbe91ab4c9048301dc4826d84ed221a5916e92fa62fdb99deef0"
+	propKey           = "TcpPortAutoRange"
+	propInstance      = "/CTRLCFG"
+)
+
+func MCValidate(w http.ResponseWriter, r *http.Request) {
+	logf.SetLogger(logr.Logger{})
+
+	arReview := v1beta1.AdmissionReview{}
+	if err := json.NewDecoder(r.Body).Decode(&arReview); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	} else if arReview.Request == nil {
+		http.Error(w, "invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	raw := arReview.Request.Object.Raw
+
+	arReview.Response = &v1beta1.AdmissionResponse{
+		UID:     arReview.Request.UID,
+		Allowed: true,
+	}
+
+	mcJson := mc{}
+	klog.Infof("Retrieving MC object %v", json.Unmarshal(raw, &mcJson))
+
+	if mcJson.Spec.Settings.DrbdPortRange.MinPort == nil && mcJson.Spec.Settings.DrbdPortRange.MaxPort == nil {
+		cl, _ := funcs.NewKubeClient()
+		ctx := context.Background()
+		objs := linstor.PropsContainersList{}
+		klog.Infof("Retrieveng list of prop containers %v", cl.List(ctx, &objs, &client.ListOptions{}))
+
+		klog.Info("Deleting port range in crd")
```

Comment:
Why are we deleting the port range in the CRD? Could lead to conflicts if LINSTOR resources use ports we remove. Maybe we should just expand the range to avoid issues and ensure smooth integration. Thoughts?